experimental/casemapping/src/internals.rs

```diff
@@ -723,29 +716,64 @@ impl<'data> CaseMapping<'data> {
         exceptions: &[u16],
         unfold: &[u16],
     ) -> Result<Self, Error> {
-        let (exceptions, valid_exception_indices) =
-            CaseMappingExceptions::try_from_icu(exceptions)?;
+        let exceptions = CaseMappingExceptions::from_raw(exceptions);
         let unfold = CaseMappingUnfoldData::try_from_icu(unfold)?;
 
         let trie_index = ZeroVec::alloc_from_slice(trie_index);
         let trie_data = trie_data
             .iter()
             .map(|&i| CaseMappingData(i))
             .collect::<ZeroVec<_>>();
-
-        for data in trie_data.iter() {
-            if data.has_exception() && !valid_exception_indices.contains(&data.exception_index()) {
-                return Error::invalid("Invalid exception index in trie data");
-            }
-        }
-
         let trie = CodePointTrie::try_new(trie_header, trie_index, trie_data)?;
 
-        Ok(Self {
+	let result = Self {
             trie,
             exceptions,
             unfold,
-        })
+        };
+	result.validate()?;
```

Comment:
Nit: You still have some tabs here.  Surprised `rustfmt` doesn't fix it for you.